### DISCUSSION TITLE
Invalid required option set for inputs in reusable workflow

### DISCUSSION THREAD

**Discussion by mritd** [community/community#55150] [Actions] [answered]
### Select Topic Area

Bug

### Body

I created a reusable workflow and set required to true in the inputs, but the reusable workflow still runs when I pass in a null value when calling.

<img width="778" alt="image" src="https://github.com/community/community/assets/13043245/ab2611ed-03ee-4b0a-ba63-fa23cf44a1f1">

<img width="996" alt="image" src="https://github.com/community/community/assets/13043245/df1a8000-900c-4aa7-954e-803ec4f0ea31">

Relevant code can access the public repository:  https://github.com/mritd/autobuild/tree/b80fe61ff5b9ef7cc0596b22c93ff6698433e175

- earthly.yaml for reusable workflow
- alipne.yaml as caller

By the way, I don't know why `${{ github.job }}` variable in `alpine.yaml` can't be used

**Comment by airtower-luna** [comment, marked as answer]
> By the way, I don't know why ${{ github.job }} variable in alpine.yaml can't be used

It's a [documented limitation for `github.job`](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context):

>  Note: This context property is set by the Actions runner, and is only available within the execution `steps` of a job. Otherwise, the value of this property will be `null`.

And technically your calling workflow does set `build-dir` to a string, the value just happens to be an empty string because `github.job` isn't available (and [`null` gets cast to an empty string](https://docs.github.com/en/actions/learn-github-actions/expressions#functions)).

**Comment by mritd** [comment]
Should an empty string check be added to `required`? `required` means that the field needs to be set mandatory, so empty strings should obviously not be the expected behavior.

**Comment by airtower-luna** [comment]
I don't know. The `required` thing is all over the examples, but neither [the syntax description](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs) nor the [Reusing Workflows article](https://docs.github.com/en/actions/using-workflows/reusing-workflows) describe how it's supposed to work. The syntax description even lacks a dedicated section, while `on.workflow_call.inputs.<input_id>.type` has one. So I'm assuming the loosest possible meaning. :sweat_smile:

**Reply by mritd** [reply]
I think GitHub's documentation seems to be very loose, and the description is not very clear... I often search for a long time and can't find what I want...

Anyway, thanks for your reply. 😄
